Kernel/System/ProcessManagement/TransitionAction/AppointmentUpdate.pm

```diff
@@ -0,0 +1,239 @@
+        . " TransitionAction: $Param{TransitionActionEntityID} - ";
+
+    # check for missing or wrong params
+    my $Success = $Self->_CheckParams(
+        %Param,
+        CommonMessage => $CommonMessage,
+    );
+    return if !$Success;
+
+    # override UserID if specified as a parameter in the TA config
+    $Param{UserID} = $Self->_OverrideUserID(%Param);
+
+    # special case for DynamicField UserID, convert form DynamicField_UserID to UserID
+    if ( defined $Param{Config}->{DynamicField_UserID} ) {
+        $Param{Config}->{UserID} = $Param{Config}->{DynamicField_UserID};
+        delete $Param{Config}->{DynamicField_UserID};
+    }
+
+    # use ticket attributes if needed
+    $Self->_ReplaceTicketAttributes(%Param);
+    $Self->_ReplaceAdditionalAttributes(%Param);
+
+    # get calendar id
+    if ( $Param{Config}->{CalendarName} ) {
+        my %Calendar = $CalendarObject->CalendarGet(
+            CalendarName => $Param{Config}->{CalendarName},
+            UserID       => 1,
+        );
+
+        $Param{Config}->{CalendarID} = $Calendar{CalendarID};
+    }
+
+    # convert some parameters to array if given
+    PARAM:
+    for my $Param (qw(TeamID ResourceID RecurrenceFrequency RecurrenceExclude )) {
+        next PARAM if !defined $Param{Config}->{$Param};
+
+        $Param{Config}->{$Param} = split /\s*,\s*/, $Param{Config}->{$Param};
+    }
+
+    # be sure that the date parameters are always in the correct format
+    PARAM:
+    for my $Param (
+        qw( StartTime EndTime RecurrenceUntil RecurrenceID RecurrenceExclude NotificationTime NotificationCustomDateTime )
+        )
+    {
+        next PARAM if !defined $Param{Config}->{$Param};
+
+        if ( ref $Param{Config}->{$Param} eq 'ARRAY' ) {
+            for my $Index ( 0 .. $#{ $Param{Config}->{$Param} } ) {
+                $Param{Config}->{$Param}->[$Index] = $Self->_ValidDateTimeConvert(
+                    String => $Param{Config}->{$Param}->[$Index],
+                );
+            }
+
+            next PARAM;
+        }
+
+        $Param{Config}->{$Param} = $Self->_ValidDateTimeConvert(
+            String => $Param{Config}->{$Param},
+        );
+    }
+
+    if (!$Param{Config}->{AppointmentID}) {
+        $Kernel::OM->Get('Kernel::System::Log')->Log(
+            Priority => 'error',
+            Message  => 'Need AppointmentID to update appointment!'
+        );
+        return;
+    }
+
+    # get dynamic field config
+    $Success = $AppointmentObject->AppointmentUpdate(
+        AppointmentID => $Param{Config}->{AppointmentID},
+        UserID => $Param{UserID},
+        %{ $Param{Config} },
+    );
+
+    return $Success;
+}
+
+=head2 _ValidDateTimeConvert()
+
+Converts the date to always be a datetime format.
+
+    my $TimeStamp = $AppointmentUpdateActionObject->_ValidDateTimeConvert(
+        String => '2019-01-01',
+    );
+
+Returns:
+
+    my $TimeStamp = '2019-01-01 00:00:00';
+
+=cut
+
+sub _ValidDateTimeConvert {
```

Comment:
The function has now been defined twice.
In 
**Kernel::System::ProcessManagement::TransitionAction::AppointmentCreate**
and 
**Kernel::System::ProcessManagement::TransitionAction::AppointmentUpdate.**

So this should now be moved to **Kernel::System::ProcessManagement::TransitionAction::Base;**.